bin/runtime/src/migrations/custom_contracts_migration.rs

```diff
@@ -0,0 +1,306 @@
+use codec::{Decode, DecodeAll, Encode, Error};
+use frame_support::{
+    codec, log,
+    pallet_prelude::*,
+    storage_alias,
+    traits::{Currency, Get, OnRuntimeUpgrade},
+    weights::Weight,
+    Twox64Concat,
+};
+use pallet_contracts::{Config, Pallet};
+use sp_runtime::traits::Saturating;
+use sp_std::{marker::PhantomData, prelude::*};
+
+use crate::sp_api_hidden_includes_construct_runtime::hidden_include::StoragePrefixedMap;
+
+type CodeHash<T> = <T as frame_system::Config>::Hash;
+type TrieId = BoundedVec<u8, ConstU32<128>>;
+
+type BalanceOf<T> =
+    <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+const TARGET: &str = "runtime::custom_contract_migration";
+
+/// Performs all necessary migrations based on `StorageVersion`.
+pub struct CustomMigrateToV8<T: Config>(PhantomData<T>);
+impl<T: Config> OnRuntimeUpgrade for CustomMigrateToV8<T> {
+    fn on_runtime_upgrade() -> Weight {
+        let version = StorageVersion::get::<Pallet<T>>();
+        let mut weight = Weight::zero();
+        log::info!(
+            target: TARGET,
+            "On-chain version of pallet contracts is {:?}",
+            version
+        );
+        if version < 8 {
```

Comment:
First of all, we don't care what happens if someone runs this migration on some old version. It will break their chain most likely :P And yes, there are older version which had different `ContractInfo` type.

In our case, the storage is in version `7` (or actually somewhere in between 7 and 8 :P). However, there is no `StorageVersion` key on chain, and hence we will see the version as `0`.